### ISSUE TITLE
should have more default loggers

### ISSUE THREAD

**Issue by ExpandingMan** [JuliaLang/julia#29567]
Some common use cases include:
- Logging progressively to a file (in which case `flush!` should be called after each log entry, depending on the output buffer).
- Logging to both `stdout` and a filer or other IO stream simultaneously (ideally it would be nice to easily be able to multiplex arbitrarily many loggers in a `MultiLogger` or some such thing).

See [this](https://discourse.julialang.org/t/how-to-save-logging-output-to-a-log-file/14004) discourse post.

We probably should also improve the logging documentation so people can get started more quickly on the common use cases.  Right now it takes a little time to figure out.

**Comment by c42f** [comment]
Yes, these things should definitely exist in the stdlib `Logging`. However I'd like to see a really solid solution in one or two non-stdlib packages first. The new logging stuff in 0.7 was intentionally minimal to avoid rushing out a half baked solution to these things.

**Comment by timholy** [comment]
If this is what you mean, [Revise is one example](https://github.com/timholy/Revise.jl/blob/master/src/logging.jl) of a package with a custom logger. But it's all internal and not focused on logging in general.

**Comment by timholy** [comment]
> What's the current state of allowing stdlib packages to be upgraded independently of julia itself?

I think it's still being decided. In general I have the feeling that we want to be cautious about breaking even the stdlibs unless we really have to.

As far as the technical side of developing a stdlib outside of base goes, Pkg.jl is the only one I know of that's actively doing that now. You might check out their infrastructure.

**Comment by c42f** [comment]
It looks like Pkg development is integrated by cherry picking commits from Pkg.jl. https://github.com/JuliaLang/Pkg.jl/issues/628 suggests there's still some work to do before we can split out stdlibs. So for now it's probably easiest to prototype elsewhere and integrate the good parts here.

I definitely don't want to break compatibility unless necessary which is exactly the reason I'm bringing up the unsolved design issues. I think building a lot of *features* into stdlib/Logging right now could be a mistake because not all the right abstractions are available yet. So I'd like to encourage experimentation in packages and we can see which abstractions work well for the backend.
